ballerina/common_utils.bal

```diff
@@ -83,6 +83,22 @@ isolated function getInvalidArgumentValueError(string listIndexOfError, string e
     }
 }
 
+isolated function getCycleRecursiveFragmentError(string spreadName, map<parser:FragmentNode> visitedSpreads)
+    returns ErrorDetail {
+    Location[] locations = [];
+    string[] keys = visitedSpreads.keys();
+    int listIndex = <int>keys.indexOf(spreadName);
```

Comment:
Shall we use `value:ensureType` rather than casting ?

Reply:
Since we already know that `spreadName` is included in the `keys` array, is that ok to directly casting here? Otherwise `value:ensureType` returns `int|error` and we need to handle the error explicitly. don't we? WDYT @ThisaruGuruge

Reply:
> Since we already know that `spreadName` is included in the `keys` array, is that ok to directly casting here? Otherwise `value:ensureType` returns `int|error` and we need to handle the error explicitly. don't we? WDYT @ThisaruGuruge

+1 to cast.